daemon/volumes.go

```diff
@@ -80,7 +81,7 @@ func (m *Mount) initialize() error {
 	if hostPath, exists := m.container.Volumes[m.MountToPath]; exists {
 		// If this is a bind-mount/volumes-from, maybe it was passed in at start instead of create
 		// We need to make sure bind-mounts/volumes-from passed on start can override existing ones.
-		if !m.volume.IsBindMount && m.from == nil {
+		if (!m.volume.IsBindMount && !m.isBind) && m.from == nil {
```

Comment:
part of the problem why this needs to be refactored  cause this looks weird but....

We have m.volume.IsBindMount, which is set if the volume itself is a bind-mount, which in the tested case it is not since it was created as a normal volume.

m.IsBind is (now) set when it is created during parsing HostConfig.Binds.